grpc-ingest/src/grpc.rs

```diff
@@ -119,97 +131,140 @@ pub async fn run(config: ConfigGrpc) -> anyhow::Result<()> {
+                                            ..Default::default()
+                                        })
+                                        .await;
+
+                                    match ping {
+                                        Ok(_) => {
+                                            debug!(target: "grpc2redis", action = "send_ping", message = "Ping sent successfully", id = PING_ID)
+                                        }
+                                        Err(err) => {
+                                            warn!(target: "grpc2redis", action = "send_ping_failed", message = "Failed to send ping", ?err, id = PING_ID)
+                                        }
+                                    }
+                                }
+                                Some(UpdateOneof::Pong(pong)) => {
+                                    if pong.id == PING_ID {
+                                        debug!(target: "grpc2redis", action = "receive_pong", message = "Pong received", id = PING_ID);
+                                    } else {
+                                        warn!(target: "grpc2redis", action = "receive_unknown_pong", message = "Unknown pong id received", id = pong.id);
+                                    }
+                                }
+                                _ => {
+                                    warn!(target: "grpc2redis", action = "unknown_update_variant", message = "Unknown update variant", ?msg.update_oneof)
+                                }
                             }
                         }
-                    }
-                    Some(UpdateOneof::Pong(pong)) => {
-                        if pong.id == PING_ID {
-                            debug!(target: "grpc2redis", action = "receive_pong", message = "Pong received", id = PING_ID);
-                        } else {
-                            warn!(target: "grpc2redis", action = "receive_unknown_pong", message = "Unknown pong id received", id = pong.id);
+                        _ = shutdown.next() => {
+                            exec.push(GrpcJob::FlushRedisPipe);
+                            break;
                         }
                     }
-                    _ => {
-                        warn!(target: "grpc2redis", action = "unknown_update_variant", message = "Unknown update variant", ?msg.update_oneof)
-                    }
                 }
-            }
-            _ = shutdown.next() => {
-                exec.push(GrpcJob::FlushRedisPipe);
-                break;
-            }
-        }
+
+                exec.join_async().await;
+
+                Ok(())
+            },
+        );
+
+        subscription_tasks.push(subscription_task_handle);
     }
 
-    exec.join_async().await;
+    // Wait for all subscription tasks to finish
+    let task_results = futures::future::join_all(subscription_tasks).await;
```

Comment:
You should be able to collect all the results into single top level result that you can check and report on.

Reply:
done